pkg/sdk/helpers_proposal.go

```diff
@@ -0,0 +1,61 @@
+package sdk
+
+import (
+	"context"
+)
+
+// validatableOpts is just a proposal how we can remove some of the boilerplate.
+type validatableOpts interface {
+	// validateProp will be renamed if accepted.
+	// Having the name `validate` now would make all other `...Opts` structs validatableOpts because of Go nature.
+	validateProp() error
```

Comment:
I will rename it then in separate PR, because I would like to add for every ...Options struct guards enforcing this interface implementation (which seems to be separate PR material)